src/Robin/Ntlm/Message/AbstractAuthenticateMessageEncoder.php

```diff
@@ -98,6 +105,35 @@ protected function __construct(EncodingConverterInterface $encoding_converter)
     }
 
     /**
+     * Identifies the "TargetName" of the intended authentication by inspecting
+     * some of the authentication details.
+     *
+     * @param string $username The user's "username".
+     * @param string $nt_domain The domain name of the NT user authenticating.
+     * @param ServerChallenge $server_challenge The value of a decoded NTLM
+     *   server's "CHALLENGE_MESSAGE".
+     * @return string The identified "TargetName" (domain/server name) of the
+     *   NT user authenticating.
+     */
+    public function identifyTargetName($username, $nt_domain, ServerChallenge $server_challenge)
+    {
+        // If a domain name wasn't supplied, fall back to the server challenge's supplied value
+        $target_name = $nt_domain ?: $server_challenge->getTargetName();
+
+        /**
+         * If the username is in the "UPN" (Kerberos) format, the target name should be empty
+         *
+         * @link https://msdn.microsoft.com/en-us/library/windows/desktop/aa380525(v=vs.85).aspx
+         * @link http://davenport.sourceforge.net/ntlm.html#nameVariations
+         */
+        if (strpos($username, static::USER_PRINCIPAL_NAME_SEPARATOR)) {
```

Comment:
Noticed this in the doc for `strpos`

http://php.net/manual/en/function.strpos.php#80363

We're good with this "truthy" style comparison here?